### PR TITLE
Fix crash admin media list api when info is None

### PR DESCRIPTION
**Pull request by schmop**
Signed-off-by: Lars Richard <lars.richard@rocketmail.com>

Fixes https://github.com/matrix-org/synapse/issues/14536

### CODE REVIEW

**Comment by schmop** [comment]
Fixes #14536

File: synapse/storage/databases/main/room.py [hunk 1]
```diff
@@ -912,7 +912,7 @@ def _get_media_mxcs_in_room_txn(
                 event_json = db_to_json(content_json)
                 content = event_json["content"]
                 content_url = content.get("url")
-                thumbnail_url = content.get("info", {}).get("thumbnail_url")
+                thumbnail_url = (content.get("info") or {}).get("thumbnail_url")
```
Comment (DMRobertson) [inline synapse/storage/databases/main/room.py:915]: This only fixes the problem if `info` is falsey (i.e. a JSON `null`, `false`, `0` or `''`). If `info` is any other value that isn't an object (`true`, number, string, array) then the problem still endures.

Ideally, we would be rejecting events with malformed event content like this in the first place.

Comment (DMRobertson) [reply synapse/storage/databases/main/room.py:915]: I suggest doing something like

```python
info = content.get("info")
if isinstance(info, dict):
    thumbnail_url = info.get("thumbnail_url")
else:
    thumbnail_url = None
```

**Review by DMRobertson — CHANGES_REQUESTED** [review]

**Review by DMRobertson — APPROVED** [review]
Thank you!

**Pull request merged**